genshi/template/astutil.py

```diff
@@ -82,6 +81,10 @@ def visit(self, node):
             info = True
         except AttributeError:
             info = False
+        if isinstance(node, (str, int)):
```

Comment:
I assume many other Python types that appear as constants are possible here? E.g. float, bytes, bool, NoneType, complex.

I guess this counts as a bug in Python 3.9 -- or at least I would complain that a parse tree should consist of AST nodes and not random types. For now could you add the other types I've listed that could appear as constants and possibly file a bug report with Python 3.9 and see what people say?

Reply:
Or maybe this is the result of us now calling `visit` on things that were nodes before and now are just constants?

Reply:
Yes, I suspect we should just not call `visit` on these items but the current code was the easiest thing I could do to fix this. I'll try to check if I can find a more elegant solution.